### PR TITLE
docs: Improve documentation

### PR DESCRIPTION
**Pull request by nikiburggraf**
This does a few things

* Fixes Sphinx autodocumentation, which will ideally fix the read-the-docs
* Adds a ton of documentation to individual files, packages, and methods
* Updates README with useful information
* Consolidates the 2 CONTRIBUTING.md files into one

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -96,6 +96,51 @@ Unit tests with coverage:
 pytest --cov=genetic_forensic_portal tests/ --cov-report xml:coverage.xml
 ```
 
+## Developing with nox
+
+The fastest way to start with development is to use nox. If you don't have nox,
+you can use `pipx run nox` to run it without installing, or `pipx install nox`.
```
Comment (lsetiawan) [inline CONTRIBUTING.md:102]: I've added nox to the `dev` dependency.. so they should get this when setting up their conda environment. Might want to modify this small section to reflect that.

File: README.md [hunk 2]
```diff
@@ -1,23 +1,45 @@
-# genetic-forensic-portal
+# Genetic Forensic Portal
```
Comment (lsetiawan) [inline README.md:1]: Please add the `ssec` project badge, `hatch` and `license` badge.

Comment (nikiburggraf) [reply README.md:1]: Can do

File: CONTRIBUTING.md (hunk 1 above)
Comment (nikiburggraf) [reply CONTRIBUTING.md:102]: Can do

**Review by lsetiawan — APPROVED** [review]
LGTM

**Pull request merged**